engine_modules/corporation/models.py

```diff
@@ -93,19 +92,48 @@ class Meta:
 	base_corporation_slug = models.CharField(max_length=20)
 	game = models.ForeignKey(Game)
 	assets = models.SmallIntegerField()
-	historic_market = models.ForeignKey(Market)
+	market_assets = models.SmallIntegerField()
+	assets_modifier = models.SmallIntegerField(default=0)
+
+	@property
+	def corporation_markets(self):
+		return self.corporationmarket_set.all()
+
+	@property
+	def markets(self):
+		return [cm.market for cm in self.corporation_markets]
+
+	@property
+	def random_market(self):
+		return random.choice(self.markets)
+
+	@property
+	def random_corporation_market(self):
+		return random.choice(self.corporation_markets)
+
+	def get_common_corporation_market(self, c2):
+		# Returns the CorporationMaket for a common market if there is one, None if not
+		common_corporation_market = None
+		common_corporation_markets = []
+
+		c2_markets = c2.markets
+		for cm in self.corporation_markets:
```

Comment:
`[common_corporation_markets = [cm for cm in self.corporation_markets if cm.market in c2_markets]`

looks more pythonic to me